test/core-tests/test-files.js

```diff
@@ -0,0 +1,64 @@
+/* eslint-env mocha */
+'use strict'
+
+const bl = require('bl')
+const expect = require('chai').expect
+const Readable = require('stream').Readable
+const bs58 = require('bs58')
+
+const IPFS = require('../../src/core')
+
+describe('files', () => {
+  var ipfs
+
+  before((done) => {
+    ipfs = new IPFS(require('./repo-path'))
+    ipfs.load(done)
+  })
+
+  it('add', (done) => {
+    const buffered = new Buffer('some data')
+    var rs = new Readable()
+    rs.push(buffered)
+    rs.push(null)
+    var arr = []
```

Comment:
how is an array that is going to be populated labelled a constant? usually languages don't offer dynamic array allocation... javascripts